tests/integration/replication-4.tcl

```diff
@@ -81,6 +81,57 @@ start_server {tags {"repl external:skip"}} {
     }
 }
 
+# Test added for issue #9993: check min-slave-* feature when evaluating Lua script.
+start_server {tags {"repl external:skip"}} {
+    start_server {} {
+        # Topo: 1 master - 1 slave
+        set master [srv -1 client]
+        set master_host [srv -1 host]
+        set master_port [srv -1 port]
+        set slave [srv 0 client]
+
+        test {First server should have role slave after SLAVEOF} {
+            $slave slaveof $master_host $master_port
+            wait_replica_online $master
+        }
+
+        test {With enough good slaves, write in Lua script is accepted} {
+            $master config set min-slaves-max-lag 3
+            $master config set min-slaves-to-write 1
+            $master eval "redis.call('set','foo','bar')" 0
+            wait_for_condition 50 100 {
+                [$slave get foo] eq {bar}
+            } else {
+                fail "Write did not synced to slave"
+            }
+        }
+
+        test {No accepted write in Lua script, if min-slaves-to-write is < attached slaves} {
+            $master config set min-slaves-max-lag 3
+            $master config set min-slaves-to-write 2
+            catch {$master eval "return redis.call('set','foo','bar')" 0} e
```

Comment:
@noradomi What about scripts that read only? Should we allow them to execute?

Reply:
@anhldbk First, I should understand what `scripts that read only` is ?

1. Script that only includes READ commands. 
2. Or, script run with READ_ONLY mode, through [eval_ro](https://redis.io/commands/eval_ro) command.

With "1", the script still be executed normally because we only verify `min-slave-*` config for WRITE commands.
`Should we allow them to execute ?` -> yes, because the data is read guranteed enventually consistency between master and slaves.
  
With "2", I see there is a check for it [here](https://github.com/redis/redis/blob/unstable/src/script.c#L258), so the WRITE commands will be rejected.

Reply:
@noradomi  I guess we should add tests for `redis.call('get','foo')`. And they should pass if replication is not ready